docs/types/reference-types.rst

```diff
@@ -153,7 +153,11 @@ third-party string libraries. You can also compare two strings by their keccak25
 concatenate two strings using ``bytes.concat(bytes(s1), bytes(s2))``.
 
 You should use ``bytes`` over ``bytes1[]`` because it is cheaper,
-since ``bytes1[]`` adds 31 padding bytes between the elements. As a general rule,
+since using ``bytes1[]`` in ``memory`` adds 31 byte padding bytes between the elements. Note that in ``storage`` the
+padding is absent due to tight packing. While ``bytes1[]`` is stored in ``storage`` based on general array packing rules,
+the storage encoding of ``bytes`` depends on the length. If the length of ``bytes`` is at most 31 bytes, it is stored adjacent to the array
+length slot ``p``. If the length is more than 32 bytes, ``bytes`` is treated in the same way as an array and it's data is stored at
+``keccak256(p)``. As a general rule,
```

Comment:
```suggestion
padding is absent due to tight packing, see :ref:`bytes and string <bytes-and-string>`. As a general rule,
```

Sorry, I think it's best not to repeat and just link the section here.

Reply:
Okay that's cool too. I initially wanted to add a link, but I thought that maintainers encourage more comprehension and less references.

Nevertheless linking is better in my opinion too.

Reply:
I think it's generally nice to avoid some duplication.